sdks/python/apache_beam/testing/analyzers/perf_analysis_test.py

```diff
@@ -0,0 +1,113 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+import os
+import time
+import unittest
+
+try:
+  import apache_beam.testing.analyzers.perf_analysis as analysis
+  from apache_beam.testing.analyzers import constants
+  from apache_beam.testing.analyzers.perf_analysis_utils import is_change_point_in_valid_window
+  from apache_beam.testing.analyzers.perf_analysis_utils import is_perf_alert
+  from apache_beam.testing.analyzers.perf_analysis_utils import e_divisive
+  from apache_beam.testing.analyzers.perf_analysis_utils import validate_config
+except ImportError as e:
+  analysis = None
+
+
+@unittest.skipIf(
+    analysis is None,
```

Comment:
signal-processing-algorithms seems light-weight. I think we  can add it to the [test] dependencies.

Reply:
I prefer having a different GHA that could run these tests + change point analysis as one action.  We can install this package as part of GHA

Reply:
hmm.. ok. it might make it harder to iterate on this checker though since tests won't run on precommits.